src/mito2/src/schedule/scheduler.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::atomic::{AtomicU8, Ordering};
+use std::sync::Arc;
+
+use common_telemetry::info;
+use tokio::task::JoinHandle;
+use tokio_util::sync::CancellationToken;
+
+use crate::error::Result;
+
+pub type Job = Pin<Box<dyn Future<Output = ()> + Send>>;
+
+/// state of scheduler
+const STATE_RUNNING: u8 = 0;
```

Comment:
Why not using [Enum](https://doc.rust-lang.org/book/ch06-01-defining-an-enum.html#defining-an-enum)

Reply:
Enumerated types cannot be modified with atomic variables